lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,302 @@
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+
+        focusableBtnIndex !== null
+          ? (focusableBtnIndex === 0 ? focusLast() : focusPrev())
+          : focusLast();
+
+        evt.preventDefault();
+        break;
+      case 'Backspace':
+        if (autoComplete) {
+          setInlineStyle(false);
+        }
+        break;
+      case 'Escape':
+        setShowDropdown(false);
+        setFocusableBtnIndex(null);
+        if (!showDropdown) {
+          handleOnChange('');
+        }
+        break;
+    }
+  }
+
+  function doSearch(val: string) {
+    focusTextInput();
+    onSearch(val.trim());
+    setShowDropdown(true);
+    if (autoComplete && !value.includes(val)) {
+      setInlineStyle(true);
+    }
+  }
+
+  function focusTextInput() {
+    inputRef.current?.querySelector('input')?.focus();
+  }
+
+  function setInlineOption() {
+    const firstOptVal = matches[0].displayName;
+    if (firstOptVal.toLocaleLowerCase().indexOf(value.toLowerCase()) === 0) {
+      setPrevSearchText(value);
+      onChange(firstOptVal);
+    }
+  }
+
+  // Nullify focusableBtnIndex whenever the number of matches changes
+  useEffect(function() {
+    if (matches.length && autoComplete && inlineStyle) {
+      setInlineOption();
+    }
+    else {
+      setFocusableBtnIndex(null);
+    }
+  }, [matches]);
+
+  useEffect(function() {
+    if (prevSearchText && autoComplete && showDropdown && inlineStyle) {
+      inputRef.current?.querySelector('input')?.setSelectionRange(prevSearchText.length, value.length);
+      focusFirst();
+    }
+  }, [prevSearchText]);
+
+  function handleOnClick({displayName}: DataItem<T, string>) {
+    onChange(displayName);
+    focusTextInput();
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  return (
+    /*eslint-disable-next-line jsx-a11y/no-static-element-interactions*/
+    <div ref={ref}
+         className={className}
+         onFocus={handleComponentFocus}
+         onBlur={handleComponentBlur}
+         { ...attrs }>
+      <NxTextInput role="combobox"
+                   ref={inputRef}
+                   id={inputId}
+                   validationErrors={validationErrors}
+                   validatable={validatable}
+                   isPristine={!!isPristine}
+                   className='nx-combobox__input'
+                   value={value}
+                   onChange={handleOnChange}
+                   disabled={disabled || undefined}
+                   onKeyDown={handleKeyDown}
+                   aria-autocomplete={autoComplete ? 'both' : 'list'}
+                   aria-expanded={showDropdown && !isAlert}
+                   aria-controls={dropdownId}
+                   aria-activedescendant={focusableBtnIndex !== null ?
+                     dropdownRef.current?.children[focusableBtnIndex].id : undefined }
+                   aria-required={ariaRequired}
+                   aria-describedby={ isAlert ? alertDropdownId : ariaDescribedBy}
```

Comment:
`aria-describedby` takes a space-separated list of ids.  When there is an alert _and_ the calling code has specified an `aria-describedby`, the two should be combined together.  Since it's a space-separate string much like a class list, the `classnames` utility is convenient for this.

Reply:
Updated here c18b4ffa478e64baa031b10867db13fb869c3294.